src/saml2/mdstore.py

```diff
@@ -616,9 +614,8 @@ def parse(self, xmlstr):
         try:
             self.entities_descr = md.entities_descriptor_from_string(xmlstr)
         except Exception as e:
-            logger.error(f'Metadata Parse Error on: {self.filename}')
-            return
-        
+            raise SAMLError(f'Failed to parse metadata file: {self.filename}') from e
```

Comment:
This is blocking ...!
A corrupted metadata file would block the entire platform. Are you sure to want this?

Reply:
Yes, `parse` either does its job or fails. The way to fail is to raise an exception.
The caller can decide if it cares to react to that, or ignore the error.